demo/lib/topology2qemuopts.py

```diff
@@ -197,10 +226,23 @@ def qemuopts(numalist):
     validate(numalist)
 
     # Read cpu counts, and "mem" and "nvmem" sizes for all nodes.
+    threadcount = -1
     for numalistindex, numaspec in enumerate(numalist):
         nodecount = int(numaspec.get("nodes", 1))
         groupnodes[numalistindex] = tuple(range(lastnode + 1, lastnode + 1 + nodecount))
-        threadcount = int(numaspec.get("threads", 2)) # threads per cpu
+        corecount = int(numaspec.get("cores", 0))
+        if corecount > 0:
+            if threadcount == -1:
+                # threads per cpu, set only once based on the first cpu-ful numa node
+                threadcount = int(numaspec.get("threads", 2))
+                threads_set_node = numaspec
+            else:
+                # threadcount already set, only check that there is no mismatch
+                if (not numaspec.get("threads", None) is None and
```

Comment:
nit: I think `is not None` would be more python-esque and possibly easier to read(?)

Reply:
Thanks, fixed! And changed other `not X is None`'s too.